### ISSUE TITLE
`S412` false negatives

### ISSUE THREAD

**Issue by ntBre** [astral-sh/ruff#19242]
[suspicious-httpoxy-import (S412)](https://docs.astral.sh/ruff/rules/suspicious-httpoxy-import/#suspicious-httpoxy-import-s412) has false negatives, as uncovered in #19009. The rule currently only applies to `ImportFrom` statements, not calls to the insecure code:

```python
from wsgiref.handlers import CGIHandler  # S412
from twisted.web.twcgi import CGIScript  # S412

import wsgiref.handlers
import twisted.web.twcgi

wsgiref.handlers.CGIHandler()  # False negative?
twisted.web.twcgi.CGIScript()  # False negative?
```

These are caught by the upstream linter:

```shell
$ cat <<EOF | uvx --with flake8-bandit flake8 --select S -
from wsgiref.handlers import CGIHandler  # S412
from twisted.web.twcgi import CGIScript  # S412

import wsgiref.handlers
import twisted.web.twcgi

wsgiref.handlers.CGIHandler()  # False negative?
twisted.web.twcgi.CGIScript()  # False negative?
EOF
Unable to find qualified name for module: stdin
stdin:1:1: S412 Consider possible security implications associated with CGIHandler module.
stdin:2:1: S412 Consider possible security implications associated with CGIScript module.
stdin:7:1: S412 Consider possible security implications associated with wsgiref.handlers.CGIHandler module.
stdin:8:1: S412 Consider possible security implications associated with twisted.web.twcgi.CGIScript module.
```

The documentation is also incorrect, which led to the initial PR. This is the example in the [docs](https://docs.astral.sh/ruff/rules/suspicious-httpoxy-import/#example):

```python
import wsgiref.handlers.CGIHandler
```

which is not flagged because it's not an `ImportFrom`. I also don't think it's a valid import because you can't import a class like that, but I can't demonstrate for `wsgiref` because `uvx --with wsgiref python` fails with errors about Python-2-style `print` statements. That's probably its own issue. Along those lines, the PyPI page for [wsgiref](https://pypi.org/project/wsgiref/) says it supports Python <3.2, so maybe we should consider deprecating that part of the rule as outdated.

## Possible solutions

A simple solution would just be to flag any import from `wsgiref.handlers`. Based on the [docs](https://peak.telecommunity.com/wsgiref_docs/module-wsgiref.handlers.html), I think everything in that module is related to CGI and could fall under the rule, not just `CGIHandler`. I think the same is true for [`twisted.web.twcgi`](https://github.com/twisted/twisted/blob/trunk/src/twisted/web/twcgi.py), which seems to be under active development still.

A more complicated solution would be to check for calls like the upstream linter. If we decide to go this route, we may need to assess other `flake8-bandit` rules for similar issues. 

However, after writing this up, I'm pretty strongly leaning toward just flagging any import, which is in line with our other `flake8-bandit` rules.

cc @MeGaGiGaGon who found the initial problem and helped track down the root of the issue

**Comment by MeGaGiGaGon** [comment]
I think `wsgiref` is referring to the [builtin python module](https://docs.python.org/3/library/wsgiref.html#wsgiref.handlers.CGIHandler), or maybe it got added to the standard library at some point? Since it works fine for me if I don't have `wsgiref` from `PyPI` installed:
```
PS ~>uvx python -c "import wsgiref.handlers.CGIHandler"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
    import wsgiref.handlers.CGIHandler
ModuleNotFoundError: No module named 'wsgiref.handlers.CGIHandler'; 'wsgiref.handlers' is not a package
PS ~>uvx python -c "from wsgiref.handlers import CGIHandler"
PS ~>   
```

**Comment by ntBre** [comment]
That explains a lot, thanks.
